devito/types/sparse.py

```diff
@@ -587,7 +593,7 @@ def gridpoints(self):
         for coords in self.coordinates.data._local:
             ret.append(tuple(int(sympy.floor((c - o.data)/i.spacing.data)) for c, o, i in
                              zip(coords, self.grid.origin, self.grid.dimensions)))
-        return ret
+        return tuple(ret)
```

Comment:
NOTE: necessary to return a tuple instead of list for hashing in memoizization (how do you spell that!!!)